nilearn/datasets/utils.py

```diff
@@ -357,6 +354,10 @@ def _uncompress_file(file_, delete_archive=True, verbose=1):
         if not processed:
             raise IOError(
                     "[Uncompress] unknown archive file format: %s" % file_)
+        else:
+            if delete_archive and os.path.isfile(file_):
+                os.remove(file_)
+
```

Comment:
+ change condition in line 351
`if os.path.is_file(file_) and tarfile.is_tarfile(file_):`